components/View/index.jsx

```diff
@@ -76,13 +80,18 @@ class View extends Component {
    * @param {Object} nextProps The new component props.
    */
   componentWillReceiveProps(nextProps) {
-    if (nextProps.title !== this.props.title) {
-      this.props.setTitle(nextProps.title || this.props.title);
-    }
-
-    if (nextProps.viewTop && nextProps.viewTop !== this.props.viewTop) {
-      // Scroll to top
-      this.element.scrollTop = 0;
+    // Only run the logic, if the path for the View instance is active
+    if (nextProps.historyPathname === this.pathname) {
```

Comment:
The use of a guard clause here would make this block easier to read.

Reply:
done